openapi-integration-tests/src/test/java/io/ballerina/openapi/OpenAPITest.java

```diff
@@ -0,0 +1,99 @@
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.AfterTest;
+import org.testng.annotations.BeforeClass;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Comparator;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * OpenAPI command cmd common class to handle temp dirs and outputs.
+ */
+public abstract class OpenAPITest {
+    protected Path tmpDir;
+    protected PrintStream printStream;
+    private ByteArrayOutputStream console;
+    private static final Path RES_DIR = Paths.get("src/test/resources/").toAbsolutePath();
+
+
+    @BeforeClass
+    public void setup() throws IOException {
+        TestUtil.cleanDistribution();
+        this.tmpDir = Files.createTempDirectory("openapi-cmd-test-out-" + System.nanoTime());
+        this.console = new ByteArrayOutputStream();
+        this.printStream = new PrintStream(this.console);
+    }
+
+    @AfterClass
+    public void cleanup() throws IOException {
+        Files.walk(this.tmpDir)
+                .sorted(Comparator.reverseOrder())
+                .forEach(path -> {
+                    try {
+                        Files.delete(path);
+                    } catch (IOException e) {
+                        Assert.fail(e.getMessage(), e);
+                    }
+                });
+        this.console.close();
+        this.printStream.close();
+        deleteGeneratedFiles();
+        TestUtil.cleanDistribution();
+    }
+
+    @AfterTest
+    public void removeGeneratedFile() {
+        deleteGeneratedFiles();
+    }
+
+    // Delete generated service and schema files.
+    public void deleteGeneratedFiles() {
+        File schemaFile = new File(this.tmpDir.resolve("types.bal").toString());
+        schemaFile.delete();
+    }
+
+    private static String getStringFromGivenBalFile(Path expectedServiceFile) throws IOException {
+        Stream<String> expectedServiceLines = Files.lines(expectedServiceFile);
+        String expectedServiceContent = expectedServiceLines.collect(Collectors.joining(System.lineSeparator()));
+        expectedServiceLines.close();
+        return expectedServiceContent.trim().replaceAll("\\s+", "").replaceAll(System.lineSeparator(), "");
+    }
+
+
+    public void compareGeneratedSyntaxTreewithExpectedSyntaxTree(String balfile) throws IOException {
+        String expectedBallerinaContent = getStringFromGivenBalFile(RES_DIR.resolve(
+                "service/return/ballerina").resolve(balfile));
+        String generatedFile = getStringFromGivenBalFile(this.tmpDir.resolve("types.bal"));
+        generatedFile = (generatedFile.trim()).replaceAll("\\s+", "");
+        expectedBallerinaContent = (expectedBallerinaContent.trim()).replaceAll("\\s+", "");
+        Assert.assertTrue(generatedFile.contains(expectedBallerinaContent));
```

Comment:
We came to some scenarios even though it has the same content, it won't work for equals since some different value white space and so on task,  but it work for contains.

Reply:
In that case, Can't we use the `trim()` on the string content and then compare using `assertEquals()`?

Reply:
In line 94, 95, we did the trim as well and remove white space as well, this issue happens interminante.